### PR TITLE
[ncl] Standalone, vanilla React Native NCL

### LINKED ISSUE #3031: Investigate whether native-component-list can be both Expo and bare RN app
Kind of a proposal.
It might be cool to have `native-component-list` that can work as:

- Expo app that runs in Expo Client
- bare React Native app with universal modules

This would allow us to test the same functionalities across both environments and not having two separate apps.

cc @brentvatne

### PR DESCRIPTION
**Pull request by sjchmiela**
# Why

Fixes https://github.com/expo/expo/issues/3031. Given our recent effort to move all Expo APIs to universal modules, we would also like to be able to easily test them from inside of `expo` monorepo. This pull request adds a “bare”, “vanilla” React Native application in `apps/standalone-ncl` that (on JS side) actually just imports and registers our regular `native-component-list` as the entrypoint. 

# How

Created a vanilla RN app, configured to work inside the monorepo, linked and setup `react-native-{screens, gesture-handler, reanimated}` and `expo-{core, react-native-adapter}`. For NCL to work properly, I added the following modules aliases: `~expo -> expo` and `expo -> ./expoResolver`. This will let us mimic `expo` package behavior in NCL without losing the ability to test `expo` in Expo NCL. We will mock/export modules one by one when integrating universal modules into vanilla NCL.

I willingly didn't integrate any universal modules into the app to allow adding-testing-fixing modules to this app one-by-one.

# Test plan

- `yarn start` in `apps/standalone-ncl`
- `pod install` in `apps/standalone-ncl/ios` (I didn't commit `Pods`, but committed `Podfile`)
- run native Xcode project
- run native Android Studio project

obrazek png 2

### CODE REVIEW

**Comment by brentvatne** [comment]
neat, this approach looks good for now

**Review by brentvatne — APPROVED** [review]

**Review by ide — CHANGES_REQUESTED** [review]
Please revert the changes to handle missing native modules -- it makes the TS types nullable for this one edge case and adds extra if-statements that should normally never fail as long as the native code is properly installed. It's also too easy to break by introducing code that assumes the corresponding native code is installed.

Projects using bare RN should use the unimodules directly instead of the `expo` package (until the `expo` package is just the core).

File: packages/expo-localization/src/ExpoLocalization.ts [hunk 1]
```diff
@@ -1,3 +1,3 @@
 import { NativeModulesProxy } from 'expo-core';
 
-export default NativeModulesProxy.ExpoLocalization;
+export default NativeModulesProxy.ExpoLocalization || {};
```
Comment (ide) [inline packages/expo-localization/src/ExpoLocalization.ts]: Still needed?

**Review by EvanBacon — APPROVED** [review]

File: packages/expo-localization/src/ExpoLocalization.ts (hunk 1 above)
Comment (sjchmiela) [reply packages/expo-localization/src/ExpoLocalization.ts]: Removed ✅

**Pull request merged**